OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -953,6 +953,16 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				if (engineVersion < 20170831)
+				{
+					if (node.Key.StartsWith("PlayerPaletteFromCurrentTileset"))
+					{
+						node.Value.Nodes.Add(new MiniYamlNode("Tileset", ""));
+						node.Key = "PaletteFromFile";
```

Comment:
Shouldn't this carry over the old suffix?

Reply:
Yes, by just renaming the node key it should exactly do that.

Note: I haven't tested this upgrade rule as this trait isn't used anymore.

Reply:
Wait you mean suffix.

Reply:
You could only have one, so it will most likely not have a suffix.

Reply:
Then give it one because this might end up conflicting.

Reply:
I found the `RenameNodeKey` function which should take care of this now.